drivers/can/mcp2515.c

```diff
@@ -0,0 +1,618 @@
+/*
+ * Copyright (c) 2018 Karsten Koenig
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <spi.h>
+#include <gpio.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_CAN_LEVEL
+#include <logging/sys_log.h>
+
+#include "mcp2515.h"
+
+static int mcp2515_cmd_soft_reset(struct device *dev)
+{
+	u8_t cmd_buf[1] = { MCP2515_OPCODE_RESET };
+
+	const struct spi_buf tx_buf = {
+		.buf = cmd_buf, .len = 1,
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf, .count = 1
+	};
+
+	return spi_write(DEV_DATA(dev)->spi, &DEV_DATA(dev)->spi_cfg, &tx);
+}
+
+static int mcp2515_cmd_bit_modify(struct device *dev, u8_t reg_addr, u8_t mask,
+				  u8_t data)
+{
+	u8_t cmd_buf[4] = { MCP2515_OPCODE_BIT_MODIFY, reg_addr, mask, data };
```

Comment:
Could you write
u8_t cmd_buf[] = { MCP2515_OPCODE_BIT_MODIFY, reg_addr, mask, data };